From e8231d88cfd89dc94482fd1ed6e4a83af3aad43e Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Thu, 24 Apr 2014 11:12:37 +0800
Subject: [PATCH 5/6] ENGR00283484 Multiuser:owner or user info always show 2
 in nickname field

In some case(e.g.without install GMS),create one multimuer,then
fill the config information. If you added phone number, you will see
the NickName in quick checking UI is always 2 not your setted name.
Trace codes and find two DB insert:
--insertInTransaction: uri=content://com.android.contacts/profile/data
values=[data1=1 234-56 data2=2 mimetype=vnd.android.cursor.item/phone_v2
raw_contact_id=9223372034707292161]
--insertInTransaction: uri=content://com.android.contacts/profile/data
values=[mimetype=vnd.android.cursor.item/name data1=Testy data4=null data5=null
data2=Testy data3=null data6=null raw_contact_id=9223372034707292161]
So in fact, it get 2 from mimetype(phone). Add a given type to let it
query mimetype(data) will work OK.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 src/com/android/settings/Utils.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/settings/Utils.java b/src/com/android/settings/Utils.java
index a8d0e8f..39510f6 100644
--- a/src/com/android/settings/Utils.java
+++ b/src/com/android/settings/Utils.java
@@ -548,7 +548,9 @@ public class Utils {
                 Profile.CONTENT_URI.buildUpon().appendPath(Contacts.Data.CONTENT_DIRECTORY).build(),
                 new String[] {CommonDataKinds.StructuredName.GIVEN_NAME,
                     CommonDataKinds.StructuredName.FAMILY_NAME},
-                Data.RAW_CONTACT_ID + "=" + localRowProfileId,
+                Data.RAW_CONTACT_ID + "=" + localRowProfileId + " AND " +
+                Data.MIMETYPE + "='" +
+                CommonDataKinds.StructuredName.CONTENT_ITEM_TYPE + "'",
                 null, null);
         if (structuredName == null) return null;
 
-- 
1.8.0

